src/App.jsx

```diff
@@ -1,17 +1,44 @@
 import './App.css';
+import { useState } from 'react';
+import messagesData from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messageList, setMessageList] = useState(messagesData);
+
+  const likeToggleFunc = (messageId) => {
+    const messages = messageList.map(message => {
+      if (message.id === messageId) {
+        return { ...message, liked: !message.liked };
+      } else {
+        return message;
+      }
+    });
+
+    setMessageList(messages);
+  };
+
+  const totalLikes = messageList
+    .filter((message) => message.liked)
+    .length;
```

Comment:
You could also do this with the[`.reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) method like so:

```javascript
const  totalLikes  =  chatMessages.reduce((count,  message)  => {  
 return count  + (message.liked ? 1 : 0)
},  0);
```
As for the formatting, I think you could leave this code on a single since it isn't longer than 80 characters.
```suggestion
  const totalLikes = messageList.filter((message) => message.liked).length;
```